contrib/coredns/policy/policytap/policytap_msg.go

```diff
@@ -0,0 +1,84 @@
+package policytap
+
+import (
+	"io"
+	"log"
+	"strings"
+	"time"
+
+	dnstap "github.com/dnstap/golang-dnstap"
+	"github.com/golang/protobuf/proto"
+	pb "github.com/infobloxopen/themis/pdp-service"
+	"github.com/miekg/dns"
+)
+
+// ToDnstap creates PolicyHitMessage and writes it to the privided Writer
+func ToDnstap(w io.Writer, t time.Time, msg *dns.Msg, tt PolicyHitMessage_PolicyTriggerType,
+	attrs []*DnstapAttribute, r *pb.Response) error {
```

Comment:
It may be a good idea to spin off another go routine for this. So that the primary query doesn't have to wait for the DNS tap message to be sent.

Reply:
Ok, will do

Reply:
Also, it should be checking if the Send CR flag is set as per https://github.com/coredns/coredns/pull/1101

Reply:
Hm, not sure if CR flag should be checked for PolicyHit messages

Reply:
Oops. Nevermind. Confused myself.

Reply:
Done sending in a separate go routine